app/util/pre_run/environment_compliance_check.py

```diff
@@ -0,0 +1,99 @@
+import sys
+
+from util.common_util import get_latest_version, get_current_version, get_unsupported_version
+from util.analytics.application_info import ApplicationSelector
+from util.analytics.analytics import MIN_DEFAULTS
+from util.conf import JIRA_SETTINGS, CONFLUENCE_SETTINGS, BITBUCKET_SETTINGS, JSM_SETTINGS, BAMBOO_SETTINGS, \
+    CROWD_SETTINGS
+
+APPS_SETTINGS = {
+    "JIRA": JIRA_SETTINGS,
+    "CONFLUENCE": CONFLUENCE_SETTINGS,
+    "BITBUCKET":  BITBUCKET_SETTINGS,
+    "JSM": JSM_SETTINGS,
+    "BAMBOO": BAMBOO_SETTINGS,
+    "CROWD": CROWD_SETTINGS,
+}
+
+
+def check_dcapt_version():
+    latest_version = get_latest_version()
+    current_version = get_current_version()
+    unsupported_version = get_unsupported_version()
+
+    if latest_version is None:
+        print('Warning: failed to get the latest version')
+    elif unsupported_version is None:
+        print('Warning: failed to get the unsupported version')
+    elif current_version <= unsupported_version:
+        raise SystemExit(f"DCAPT version {current_version} is no longer supported. "
+                         f"Consider an upgrade to the latest version: {latest_version}")
+    elif current_version < latest_version:
+        print(f"Warning: DCAPT version {current_version} is outdated. "
+              f"Consider upgrade to the latest version: {latest_version}.")
+    elif current_version == latest_version:
+        print(f"Info: DCAPT version {current_version} is the latest.")
+    else:
+        print(f"Info: DCAPT version {current_version} "
+              f"is ahead of the latest production version: {latest_version}.")
+
+
+def validate_application_config(processors, app_name_upper, app_settings, min_defaults):
+    is_jsm = app_name_upper == "JSM"
+    if not is_jsm:
```

Comment:
Could you pls refactor lines 43-52. 
If is_jsm do this, else do that.
Also, it is worth to have currren_concurrency >= min_defualts, but not ==.